api/src/main/java/org/apache/iceberg/transforms/UnknownTransform.java

```diff
@@ -33,6 +33,14 @@ public class UnknownTransform<S, T> implements Transform<S, T> {
     this.transform = transform;
   }
 
+  /**
+   * Transforms a value to its corresponding partition value.
+   *
+   * @param value a source value
+   * @return a transformed partition value
```

Comment:
nit: This explanation looks a little misleading to me because this method always throws an exception.